### DISCUSSION TITLE
Beehave documentation feedback

### DISCUSSION THREAD

**Discussion by bitbrain** [bitbrain/beehave#142] [General]
Hello!

Feel free to give us your honest feedback on the documentation of this addon, especially these places:

1. [the official README.md](https://github.com/bitbrain/beehave#readme)
2. [the official wiki](https://bitbra.in/beehave/#/)

Things to consider:

- what do you like about the documentation? What don't you like and why?
- what could be improved on the existing documentation? Especially things like structure and ordering of things
- is there anything you would like to see on the documentation that is currently missing?

**Comment by BenjaminNavarro** [comment]
Since you mentioned this addon being node based and having a custom debug view I would suggest to put a couple of screenshots in the readme and the wiki. This would give potential users a quick idea of how things work with beehave.

In the wiki you could also put an example of a BT in its graphical form and how to write it with beehave. Or multiple small ones maybe to showcase all the available nodes.

**Comment by FonkinS** [comment]
I feel that the word _learning_ is misleading. Learning should be for actual tutorials, like the one @bitbrain made. The descriptions of each of the nodes should be in a seperate branch, called _Manual_ or _Documentation_ or just _Nodes_.

As for the actual tutorials, it would be nice if they were both in video, and written format, so at the top of the page you might have the name of the tutorial, what its teaching, some keywords based on the nodes that it covers, then a video version of the tutorial, and then a kind of transcription, teaching the _lesson_ in a written format.

**Comment by lostptr** [comment]
Is the part about `begin_run` and `end_run` missing? I could not find it...

**Reply by bitbrain** [reply]
Might have been lost in transition. We could add it to https://bitbra.in/beehave/#/action_leaf

**Reply by lostptr** [reply]
I think we need a section for all the overridable methods: `tick`, `interrupt`, `begin_run` and `end_run` as they are not exclusive to leaf nodes.

**Comment by keyinora** [comment]
Could you update the documentation on how to access the debug view?

I can't seem to find how to activate it.

**Reply by FonkinS** [reply]
The debug view was made during the documentation transition, and thus the documentation for it never existed. Maybe we can get @DarkMessiah to do it as he made the debug view, (if I remember correctly) and is thus most suited to explain it.

**Reply by DarkMessiah** [reply]
![image](https://user-images.githubusercontent.com/1756388/221863786-1a082285-3fe5-4be0-9a34-0ee8e1dd75f7.png)

**Reply by keyinora** [reply]
I have the demo setup on my computer with the addon installed and this is what I see in the console, while it is running.

![not found](https://user-images.githubusercontent.com/6598270/221884398-c6879c0d-8462-4c42-97b1-93a1c86b1331.PNG)

**Reply by DarkMessiah** [reply]
Oh, this is only for Godot 4 :'(

**Reply by bitbrain** [reply]
Godot 3 will enter maintenance mode anyways **very** soon, so worth moving to Godot 4! :)

We do not plan to migrate features to the Godot 3 of Beehave, unless someone raises a PR to do so.

**Comment by mechPenSketch** [comment]
I'd like to know how `interupt(...)` works and where should I call it (if the method is supposed to be called to begin with).

May or may not be related to `interupt(...)`, but once a `SequenceComposite` `tick(...)` one of its child and receives `SUCCESS` or `FAILURE`, it will never `tick(...)` it again. Is there a way for the `SequenceComposite` to go back to its previous child? I'm also experimenting with `SelectorReactiveComposite`, which does loops back to the previous child as I hoped.

[EDITED] Nvm I've resolved the isue, which turned out not to be related to beehave, after all.

**Comment by gamecoder-nz** [comment]
I notice the before_run and after_run is missing from the documentation. Is there an API documentation?
